<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>950</height>
   </rect>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string/>
   </property>
   <layout class="QVBoxLayout" name="main_layout">
    <property name="leftMargin">
     <number>20</number>
    </property>
    <property name="topMargin">
     <number>20</number>
    </property>
    <property name="rightMargin">
     <number>20</number>
    </property>
    <property name="bottomMargin">
     <number>20</number>
    </property>
    <item>
     <widget class="QPushButton" name="returnHome">
      <property name="maximumSize">
       <size>
        <width>90</width>
        <height>60</height>
       </size>
      </property>
      <property name="font">
        <font>
         <pointsize>11</pointsize>
        </font>
       </property>
       <property name="cursor">
        <cursorShape>PointingHandCursor</cursorShape>
       </property>
       <property name="styleSheet">
        <string notr="true">
                          QPushButton {
                              background-color: #4a9fab;
                              border: none;
                              color: white;
                              padding: 10px ;
                              margin: 10px;
                              height: 35px;
                          }
                          QPushButton:hover {
                              background-color: #3e858f;
                          }
                      </string>
       </property>
       <property name="text">
        <string>Home</string>
       </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <property name="bottomMargin">
       <number>10</number>
      </property>
      <item>
       <widget class="QLabel" name="top_title">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>150</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>16</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Shortest Path Optimization</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="left_widget">
       <layout class="QVBoxLayout" name="left_layout">
        <item>
         <layout class="QHBoxLayout" name="node_layout">
          <item>
           <widget class="QLabel" name="node_input_title">
            <property name="text">
             <string>Node Input</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="add_node_button">
            <property name="text">
             <string>Add Node</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTableWidget" name="node_input_table">
          <property name="columnCount">
           <number>5</number>
          </property>
          <column/>
          <column/>
          <column/>
          <column/>
          <column/>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="edge_layout">
          <property name="topMargin">
           <number>15</number>
          </property>
          <item>
           <widget class="QLabel" name="edge_input_title">
            <property name="text">
             <string>Edge Input</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="add_edge_button">
            <property name="text">
             <string>Add Edge</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTableWidget" name="edge_input_table">
          <property name="columnCount">
           <number>6</number>
          </property>
          <property name="topMargin" stdset="0">
           <number>20</number>
          </property>
          <column/>
          <column/>
          <column/>
          <column/>
          <column/>
          <column/>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="node_selection_layout">
          <property name="topMargin">
           <number>15</number>
          </property>
          <item>
           <widget class="QLabel" name="node_selection_label">
            <property name="text">
             <string>Node Selection</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="node_selection_layout">
          <property name="topMargin">
           <number>5</number>
          </property>
          <item>
           <widget class="QLabel" name="start_node_label">
            <property name="text">
             <string>Start Node:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="start_node_combo">
            <property name="maximumSize">
             <size>
              <width>200</width>
              <height>50</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="node_selection_layout">
          <item>
           <widget class="QLabel" name="end_node_label">
            <property name="text">
             <string>End Node:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="end_node_combo">
            <property name="maximumSize">
             <size>
              <width>200</width>
              <height>50</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="submit_buttons_layout">
          <property name="topMargin">
           <number>50</number>
          </property>
          <property name="bottomMargin">
           <number>50</number>
          </property>
          <item>
           <widget class="QPushButton" name="calculate_button">
            <property name="text">
             <string>Calculate Shortest Path</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="clear_button">
            <property name="text">
             <string>Clear</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="result_info_label">
          <property name="text">
           <string>Result Information:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="result_info_display">
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarAsNeeded</enum>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="fixedHeight" stdset="0">
           <number>50</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="error_label">
          <property name="text">
           <string>Error Messages:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="error_display">
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarAsNeeded</enum>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="fixedHeight" stdset="0">
           <number>50</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="right_widget">
       <layout class="QVBoxLayout" name="right_layout">
        <item>
         <layout class="QVBoxLayout" name="plane_layout">
          <item>
           <widget class="QLabel" name="plane_title">
            <property name="text">
             <string>Content</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTextEdit" name="plane_display">
            <property name="verticalScrollBarPolicy">
             <enum>Qt::ScrollBarAsNeeded</enum>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="graph_layout">
          <item>
           <widget class="QLabel" name="graph_title">
            <property name="text">
             <string>Graph Visualization</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer>
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Expanding</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>100</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
